src/dmd/backend/cg.d

```diff
@@ -52,3 +50,110 @@ Symbol *tls_get_addr_sym;       // function __tls_get_addr
 int TARGET_STACKALIGN = 2;      // default for 16 bit code
 int STACKALIGN = 2;             // varies for each function
 }
+
+import core.stdc.stdio;
+
+extern (C++) __gshared
+{
+    ubyte[FLMAX] datafl =
+    () {
+        ubyte[FLMAX] datafl;
+        foreach (fl; [ FLdata,FLudata,FLreg,FLpseudo,FLauto,FLfast,FLpara,FLextern,
+                       FLcs,FLfltreg,FLallocatmp,FLdatseg,FLtlsdata,FLbprel,
+                       FLstack,FLregsave,FLfuncarg,
+                       FLndp, FLfardata,
+                     ])
+        {
+            datafl[fl] = 1;
+        }
+        return datafl;
+    } ();
+
+
+    ubyte[FLMAX] stackfl =
+    () {
+        ubyte[FLMAX] stackfl;
+        foreach (fl; [ FLauto,FLfast,FLpara,FLcs,FLfltreg,FLallocatmp,FLbprel,FLstack,FLregsave,
+                       FLfuncarg,
+                       FLndp,
+                     ])
+        {
+            stackfl[fl] = 1;
+        }
+        return stackfl;
+    }();
+
+    ubyte[FLMAX] segfl =
+    () {
+        ubyte[FLMAX] segfl;
+
+        /* Segment registers    */
+        enum ES = 0;
+        enum CS = 1;
+        enum SS = 2;
+        enum DS = 3;
+
+        foreach (i;  0 .. FLMAX)
+        {   switch (i)
+            {
+                case 0:         segfl[i] = cast(byte)-1;  break;
```

Comment:
```suggestion
                case 0:         segfl[i] = byte(-1);  break;
```

Reply:
And all following occurences

Reply:
As this is a copy/paste refactoring job, I avoid making more changes in the same PR.